ci: fix compiler tests on windows

alan-agius4 opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

filipesilva commented on packages/compiler/test/BUILD.bazel:
```diff
@@ -83,7 +83,6 @@ jasmine_node_test(
     deps = [
         ":test_lib",
         ":test_node_only_lib",
-        "//packages/core",
```
Are you sure this should be removed? `packages\compiler\test\directive_lifecycle_spec.ts` uses `import {SimpleChanges} from '@angular/core';;` for instance.

alan-agius4 replied:
Isn't it why there is?
```
    data = [
        "//packages/animations:npm_package",
        "//packages/common:npm_package",
        "//packages/core:npm_package",
    ],
```

filipesilva replied:
Yes, that might be it.

alan-agius4 committed:
build: remove extra unused deps for `compiler:test`

It's unnecessary for a jasmine_node_test rule to depend on a TypeScript library. This dependency is already satisfied via the 'data' and also having it in 'deps' causes CI flakiness on Windows

alan-agius4 committed:
refactor: remove unused local imports and use spread instead of slice

This is a purly estatic change and a minor cleanup

gkalpak commented on packages/compiler/test/aot/test_util.ts:
```diff
@@ -748,7 +744,8 @@ function isDts(fileName: string): boolean {
 function isSourceOrDts(fileName: string): boolean {
-  return /\.ts$/.test(fileName);
+  return /\.ts$/.test(fileName) &&
+      !/(ngfactory|shim\.ngstyle|ngstyle|ngsummary).d.ts/.test(fileName);
```
- `shim\.ngstyle` is redundant, since it will be matched by the `ngstyle` part anyway.
- Unlikely to matter, but having a `$` at the end would be more correct 😇

alan-agius4 replied:
Why do you have to be always so right :P? Hahah

Good catches :)

gkalpak replied:
I once read a whole book on RegExps 😛

filipesilva replied:
Ah yes, the classic missing `\0`. We've all been there.

alan-agius4 committed:
test: fix compiler tests in windows ci

This is a tentative fix for the error `Cannot write file '/node_modules/@angular/core/core.ngfactory.d.ts' because it would overwrite input file.` that is showing in codefresh windows ci.

filipesilva commented:
Ran `yarn bazel test //packages/compiler/test` some 30 times with this PR using the same docker image that is used in Codefresh. None failed with the errors we were seeing before, so that's encouraging.
